consensus/leader.go

```diff
@@ -244,6 +239,11 @@ func (consensus *Consensus) onCommit(msg *msg_pb.Message) {
 		return
 	}
 
+	// Check for potential double signing
+	if consensus.checkDoubleSign(recvMsg) {
```

Comment:
So this won't simply work. Actually lots of logic reply on the fact the p2p message are verified and legit. If people now can manipulate the message since we don't check sig anymore, then people can attack.
![image](https://user-images.githubusercontent.com/1042540/87751019-ded71d80-c7b1-11ea-8e2a-d9dd204dd930.png)

For example, the block hash in the p2p message are used for double sign check.

Reply:
the block signature is verified in onCommit https://github.com/harmony-one/harmony/blob/6029ba5cf9a608aad9256b212a319d201dc4512c/consensus/leader.go#L242

that VerifyHash verified the blocknum, viewed, and hash already, so the messages verified.  As I mentioned, the p2p message level verification only have 2 additional info, msg type and service type, so why do you think the p2p messages was tampered?

Reply:
noted the commitPayload was created from `recvMsg`, which is copied from `msg` p2p data.  It is the signature verification of the p2p message payload.

Reply:
You are right. The block signature verification here actually made sure the blockHash and blocknum viewid are all good and verified. Resolved.